lambda/audit.py

```diff
@@ -73,15 +75,22 @@ def log_org_teams(message):
             )
 
 
-def log_org_team_membership(message):
+def log_org_team_membership(message: Dict[str, Any]) -> None:
     """ Audit github organization team membership """
-    org = os.environ.get("GITHUB_ORG")
-    team = message.get("team", None)
-    team_name = team["slug"]
+    org = os.environ["GITHUB_ORG"]
+
+    assert isinstance(message["team"], dict)
+    team = message["team"]
+
+    if "slug" in team and isinstance(team["slug"], str):
+        team_name: Optional[str] = team["slug"]
+    else:
+        team_name = None
+
     audit_id = message.get("audit_id")
-    if audit_id is not None:
+    if isinstance(audit_id, str):
         LOG.info({"action": "Audit team members", "org": org, "team": team_name})
-        if team is not None:
+        if isinstance(team, dict) and isinstance(team_name, str):
```

Comment:
I'm not jazzed about these, this makes the code more brittle for what gain?
we don't **really** care that `audit_id` is a string, only that it quacks like a string, right?

Reply:
I think the pythonic way to do this is to replace 
```
if audit_id is not None
```
with 
```
if audit_id
```

Reply:
if you're concerned about runtime integrety, I'd suggest `assert audit_id`. make it a contract at the top of the function?

Reply:
I've changed this in all the places it makes sense to and mypy allows it.

Reply:
I truly believe that everyone did the best job they could, given what they knew at the time, their skills and abilities, the resources available and the situation at hand.